pkg/crc/machine/bundle/metadata.go

```diff
@@ -19,39 +19,56 @@ import (
 
 type CrcBundleInfo struct {
-	Version   string `json:"version"`
-	Type      string `json:"type"`
-	BuildInfo struct {
-		BuildTime                 string `json:"buildTime"`
-		OpenshiftInstallerVersion string `json:"openshiftInstallerVersion"`
-		SncVersion                string `json:"sncVersion"`
-	} `json:"buildInfo"`
-	ClusterInfo struct {
-		OpenShiftVersion      string `json:"openshiftVersion"`
-		ClusterName           string `json:"clusterName"`
-		BaseDomain            string `json:"baseDomain"`
-		AppsDomain            string `json:"appsDomain"`
-		SSHPrivateKeyFile     string `json:"sshPrivateKeyFile"`
-		KubeConfig            string `json:"kubeConfig"`
-		KubeadminPasswordFile string `json:"kubeadminPasswordFile"`
-	} `json:"clusterInfo"`
-	Nodes []struct {
-		Kind          []string `json:"kind"`
-		Hostname      string   `json:"hostname"`
-		DiskImage     string   `json:"diskImage"`
-		KernelCmdLine string   `json:"kernelCmdLine"`
-		Initramfs     string   `json:"initramfs"`
-		Kernel        string   `json:"kernel"`
-		InternalIP    string   `json:"internalIP"`
-	} `json:"nodes"`
-	Storage struct {
-		DiskImages []struct {
-			Name   string `json:"name"`
-			Format string `json:"format"`
-			Size   string `json:"size"`
-		} `json:"diskImages"`
-	} `json:"storage"`
-	cachedPath string
+	Version     string      `json:"version"`
+	Type        string      `json:"type"`
+	BuildInfo   BuildInfo   `json:"buildInfo"`
+	ClusterInfo ClusterInfo `json:"clusterInfo"`
+	Nodes       []Node      `json:"nodes"`
+	Storage     Storage     `json:"storage"`
+	DriverInfo  DriverInfo  `json:"driverInfo"`
+	cachedPath  string
+}
+
+type BuildInfo struct {
+	BuildTime                 string `json:"buildTime"`
+	OpenshiftInstallerVersion string `json:"openshiftInstallerVersion"`
+	SncVersion                string `json:"sncVersion"`
+}
+
+type ClusterInfo struct {
+	OpenShiftVersion      string `json:"openshiftVersion"`
+	ClusterName           string `json:"clusterName"`
+	BaseDomain            string `json:"baseDomain"`
+	AppsDomain            string `json:"appsDomain"`
+	SSHPrivateKeyFile     string `json:"sshPrivateKeyFile"`
+	KubeConfig            string `json:"kubeConfig"`
+	KubeadminPasswordFile string `json:"kubeadminPasswordFile"`
+	OpenshiftPullSecret   string `json:"openshiftPullSecret,omitempty"`
+}
+
+type Node struct {
+	Kind          []string `json:"kind"`
+	Hostname      string   `json:"hostname"`
+	DiskImage     string   `json:"diskImage"`
+	KernelCmdLine string   `json:"kernelCmdLine,omitempty"`
+	Initramfs     string   `json:"initramfs,omitempty"`
+	Kernel        string   `json:"kernel,omitempty"`
+	InternalIP    string   `json:"internalIP"`
+}
+
+type Storage struct {
+	DiskImages []DiskImage `json:"diskImages"`
+}
+
+type DiskImage struct {
+	Name     string `json:"name"`
+	Format   string `json:"format"`
+	Size     string `json:"size"`
+	Checksum string `json:"sha256sum"`
+}
+
+type DriverInfo struct {
+	Name string `json:"name"`
 }
```

Comment:
This hunk would make more sense in a separate commit after the commit adding the unit test, as this is exactly the kind of change where we need the unit test to make sure we did not break anything.

Reply:
I tried but the test change completely because the struct also change.
We don't break anything here because in the unit test, I take the raw json, unmarshall, marshall and compare to the raw json given as input.

Reply:
It depends where you got the initial raw json from.  If it's the sample one from snc, then it's good, if it's one which was generated by a previous run of the test case, it's less good :)

Reply:
Took it from my local running crc installation. So, from the snc release CI.